analytics-ios-mcvid/Classes/SEGMCVIDTracker.m

```diff
@@ -61,114 +225,52 @@ - (void)sendRequestAdobeExperienceCloud:(NSString *)advertisingId organizationId
     NSURLComponents *components = [NSURLComponents componentsWithString:urlString];
     NSMutableArray *queryItems = [NSMutableArray array];
 
-    //Error handling variables
-    NSString *const MCVIDAdobeErrorKey = @"MCVIDAdobeErrorKey";
-    NSString *errorResponseKey = @"error_msg";
-    NSString *invalidMarketingCloudId = @"<null>";
-    NSString *errorDomain = @"Segment-Adobe";
-    NSString *serverErrorDomain = @"Segment-Adobe Server Response";
-
-
-    if (advertisingId) {
-      NSString *encodedAdvertisingValue = [NSString stringWithFormat:@"%@%@%@", deviceTypeKey, separator, advertisingId];
-      //removes %25 html encoding of '%'
-      NSString *normalAdvertisingValue = [encodedAdvertisingValue stringByReplacingPercentEscapesUsingEncoding:NSUTF8StringEncoding];
-      [queryItems addObject:[NSURLQueryItem queryItemWithName:advertisingIdKey value:normalAdvertisingValue]];
-    }
-
     [queryItems addObject:[NSURLQueryItem queryItemWithName:versionKey value:version]];
     [queryItems addObject:[NSURLQueryItem queryItemWithName:jsonFormatterKey value:jsonFormatter]];
     [queryItems addObject:[NSURLQueryItem queryItemWithName:regionKey value:region]];
     [queryItems addObject:[NSURLQueryItem queryItemWithName:organizationIdKey value:organizationId]];
-
+    
+    if (marketingCloudId) {
+        [queryItems addObject:[NSURLQueryItem queryItemWithName:marketingCloudIdKey value:marketingCloudId]];
+        NSString *encodedAdvertisingValue = [NSString stringWithFormat:@"%@%@%@", deviceTypeKey, separator, advertisingId];
+        //removes %25 html encoding of '%'
+        NSString *normalAdvertisingValue = [encodedAdvertisingValue stringByReplacingPercentEscapesUsingEncoding:NSUTF8StringEncoding];
```

Comment:
🚨 Another bug here: by invoking `stringByReplacingPercentEscapesUsingEncoding:NSUTF8StringEncoding` you are effectively removing the `%01` separator from `encodedAdvertisingValue`, making this value and the id-sync request invalid.